src/main/java/edu/oregonstate/cs361/battleship/Military.java

```diff
@@ -12,6 +12,28 @@ public Military(boolean b, String n, int l,Coordinate s, Coordinate e) {
         stealth = b;
     }
 
+    //New scan function that implements the stealth feature, original scan function is in Ship.java --Rebecca
+    public boolean scan(Coordinate coor) {
+        if (stealth == true) {
```

Comment:
Nice, haven't tested it yet but this looks like it would work.

Also, have you considered doing this in ``BattleshipModel``'s ``scan`` instead? That way you wouldn't have to duplicate code between ``Ship`` and ``Military``.